File: tests/ops_and_seeds_codecs.py
import json
import pathlib

from sub_byte import make_sub_byte_encoder_and_decoder

SYMBOLS = json.loads((pathlib.Path(__file__).parent / 'symbols.json').read_text())

ALL_SEEDS = [str(seed) for seed in SYMBOLS['SEEDS']]
OPS = SYMBOLS['OPS']


encode_ops, decode_ops, bits_per_op, ops_per_byte = make_sub_byte_encoder_and_decoder(symbols = OPS)

encode_seeds, decode_seeds, bits_per_seed, seeds_per_byte = make_sub_byte_encoder_and_decoder(symbols = ALL_SEEDS)

File: tests/ops_and_seeds_codecs.mjs
import SYMBOLS from "./symbols.json" //with { type: "json" };

import { MakeSubByteEncoderAndDecoder } from "../src/subByte/factories.mjs";

const ALL_SEEDS = SYMBOLS.SEEDS;

// Sets are not necessary for MakeSubByteEncoderAndDecoder (it will
// create a set itself).  They're just to make writing the
// tests a little easier.
export const UNIQUE_SEEDS = new Set(SYMBOLS.SEEDS.map((x) => x.toString()));
export const OPS = new Set(SYMBOLS.OPS);

export const [encodeOps, decodeOps, bitsPerOp, numOpsPerByte] =
  MakeSubByteEncoderAndDecoder(OPS);
export const [encodeSeeds, decodeSeeds, bitsPerSeed, numSeedsPerByte] =
  MakeSubByteEncoderAndDecoder(ALL_SEEDS);
